examples/ui_actions_explorer/public/plugin.tsx

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Plugin, CoreSetup, AppMountParameters } from 'kibana/public';
+import { UiActionsStart, UiActionsSetup } from 'src/plugins/ui_actions/public';
+import { ISearchAppMountContext } from '../../../src/plugins/data/public';
+import {
+  PHONE_TRIGGER,
+  USER_TRIGGER,
+  COUNTRY_TRIGGER,
+  createPhoneUserAction,
+  lookUpWeatherAction,
+  viewInMapsAction,
+  createEditUserAction,
+  CALL_PHONE_NUMBER_ACTION,
+  VIEW_IN_MAPS_ACTION,
+  TRAVEL_GUIDE_ACTION,
+  PHONE_USER_ACTION,
+  EDIT_USER_ACTION,
+  makePhoneCallAction,
+  showcasePluggability,
+  SHOWCASE_PLUGGABILITY_ACTION,
+} from './actions/actions';
+
+declare module 'kibana/public' {
+  interface AppMountContext {
+    search?: ISearchAppMountContext;
+  }
+}
+
+interface StartDeps {
+  uiActions: UiActionsStart;
+}
+
+interface SetupDeps {
+  uiActions: UiActionsSetup;
+}
+
+export class UiActionsExplorerPlugin implements Plugin<void, void, {}, StartDeps> {
+  public setup(core: CoreSetup<{ uiActions: UiActionsStart }>, deps: SetupDeps) {
+    deps.uiActions.registerTrigger({
+      id: COUNTRY_TRIGGER,
+      actionIds: [],
+    });
+    deps.uiActions.registerTrigger({
+      id: PHONE_TRIGGER,
+      actionIds: [],
+    });
+    deps.uiActions.registerTrigger({
+      id: USER_TRIGGER,
+      actionIds: [],
+    });
+
+    core.application.register({
+      id: 'uiActionsExplorer',
+      title: 'Ui Actions Explorer',
+      async mount(params: AppMountParameters) {
+        const [coreStart, depsStart] = await core.getStartServices();
+        deps.uiActions.registerAction(createPhoneUserAction(depsStart.uiActions));
+        deps.uiActions.registerAction(lookUpWeatherAction);
+        deps.uiActions.registerAction(viewInMapsAction);
+        deps.uiActions.registerAction(createEditUserAction(coreStart.overlays.openModal));
+        deps.uiActions.registerAction(makePhoneCallAction);
+        deps.uiActions.registerAction(showcasePluggability);
```

Comment:
Shall we register these in `start` life-cycle, or even better, in `setup` life-cycle?

Reply:
Moved what I could to setup, rest in start, a couple needed some start context.

Reply:
Not in this PR, but in general, should we register everything in *setup* life-cycles? As that was the idea for the setup life-cycle. For example, if something needs *start* life-cycle contracts we can pass in the promise:

```ts
setup(core) {
  const start = core.getStartServices();
  deps.uiActions.registerAction(createPhoneUserAction(start));
}
```

Reply:
Yes! Thanks, updated in this PR.